onnxruntime/contrib_ops/webgpu/bert/flash_attention.cc

```diff
@@ -224,12 +224,12 @@ Status FlashAttentionProgram::GenerateShaderCode(ShaderHelper& shader) const {
   // Shader is designed to be dispatched as Dispatch(num_heads, new_sequence_length / workgroup_size_x, 1)
   // Each lane/thread is responsible for a single q.
   shader.MainFunctionBody() << R"MAIN_FN(
-  let head_idx = workgroup_id.x;
+  let head_idx = workgroup_idx / uniforms.num_seq_tile;
```

Comment:
u32(workgroup_idx / uniforms.num_seq_tile)

Reply:
Both `workgroup_idx` and `uniforms.num_seq_tile` are u32, may be not needed here to explicitly add `u32` conversion?

Reply:
hmm I have run into bad bugs where head_idx ends up being a float number silently when workgroup_idx  is not divisible by uniforms.num_seq_tile. I think it would be good to enforce that the result is a u32. WDYT ?

Reply:
Thanks for the information. Make sense for me. Done.